File: OCRP-Backend/src/main/java/com/app/services/StudentService.java
package com.app.services;

import java.util.List;

import com.app.pojos.Job;
import com.app.pojos.Student;

public interface StudentService {

	public List<Student> getAllStudents();

	public Student getStudentById(int sId);

	public Student registerStudent(Student student);

	public Student updateStudent( Student student);

	public String deleteStudentById(int sId);

	public String applyJob(int sId,Job job);
	
	
	
}

File: OCRP-Frontend/src/Services/StudentService.js
import React,{Component} from "react";
import axios from "axios";

const CRS_URL = "http://localhost:9090/CRS_Backend/students";

class StudentService extends Component {

  registerStudent(newStudent) {
    return axios.post(CRS_URL, newStudent);
  }

  getStudentById(sId) {
      return axios.get(CRS_URL + "/" + sId);
  }

  updateStudentById(sId,studentDetails) {
    return axios.put(CRS_URL + "/" + sId, studentDetails);
  }

  
  getAllStudent() {
    return axios.get(CRS_URL);
  }

  deleteStudentById(sId) {
    return axios.delete(CRS_URL + "/" + sId);
  }

  applyJob(sId) {
      return axios.post(CRS_URL + "/" + sId);
  }

}

export default new StudentService();